File: 137. Single Number II.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
    let res = 0;
    for (let i = 0; i < 32; i++)
    {
        let bit = 1 << i, count = 0;
        for (let j = 0; j < nums.length; j++)
            count += bit & nums[j];
        if (count % 3 !== 0)
            res += bit;
    }
    let maxint = Math.pow(2, 31);
    if (res >= maxint)
        res -= 2 * maxint;
    return res;
};

File: 137. Single Number II.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        res = 0
        for i in xrange(32):
            count = 0
            bit = 1 << i
            for j in nums:
                count += j & bit
            if count % 3 != 0:
                res += bit
        maxint = 2 ** 31
        if res >= maxint:
            res -= 2 * maxint
        return res
